src/libLookup/Lookup.cpp

```diff
@@ -3662,7 +3659,48 @@ void Lookup::SenderTxnBatchThread() {
   DetachedFunction(1, main_func);
 }
 
-void Lookup::SendTxnPacketToNodes(uint32_t numShards) {
+void Lookup::RectifyTxnShardMap(const uint32_t oldShardSize,
+                                const uint32_t newShardSize) {
+  LOG_MARKER();
+
+  auto t_start = std::chrono::high_resolution_clock::now();
+
+  map<uint, vector<Transaction>> tempTxnShardMap;
+
+  lock_guard<mutex> g(m_txnShardMapMutex);
+
+  LOG_GENERAL(INFO, "Shard dropped or gained, shuffling txn shard map");
+  LOG_GENERAL(INFO, "New Shard Size: " << newShardSize
+                                       << "Old Shard Size: " << oldShardSize);
+  for (const auto& shard : m_txnShardMap) {
+    if (shard.first == oldShardSize) {
+      // ds txns
+      continue;
+    }
+    for (const auto& tx : shard.second) {
+      const PubKey& senderPubKey = tx.GetSenderPubKey();
+      Address fromAddr = Account::GetAddressFromPublicKey(senderPubKey);
+      unsigned int correct_shard =
+          Transaction::GetShardIndex(fromAddr, newShardSize);
```

Comment:
It would be much preferred if you can keep the shard deciding logic in a reusable API. So whenever you try to get the shard to assign to just use that API instead of drafting the algo again. Which serves for consistency among the code and also easier for future maintenance

Reply:
Okay will do in separate PR